client_test.go

```diff
@@ -163,6 +163,53 @@ func getHostAndPortFromTestServer(ts *httptest.Server) (string, int, error) {
 	return hostSplits[0], port, nil
 }
 
+func TestDeleteDatasetExisting(t *testing.T) {
+	assert := assert.New(t)
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal("DELETE", r.Method)
+		assert.Equal("/v1/configuration/datasets/uuid1", r.URL.Path)
+		w.Header().Set("Content-Type", "application/json")
+		w.Write([]byte("This is an example server.\n"))
```

Comment:
No because we are not even decoding the output. The dataset is considered as deleted when we get a 200. The dodgy live is removed...